File: LeetCode/Easy/409 - Longest Palindrome.js
/**
 * @param {string} s
 * @return {number}
 */
var longestPalindrome = function(s) {
    const fr = {};
    let ret = 0;
    for (let i of s) {
        fr[i] = fr[i] + 1 || 1;
        if (!(fr[i] & 1)) ret += 2;
    }
    return s.length > ret ? ret + 1 : ret;
};

File: LeetCode/Easy/409 - Longest Palindrome/409 - Longest Palindrome.cpp
class Solution {
public:
    int longestPalindrome(string s) {
        map<char, int> fr;
        int ret = 0;
        for (auto c : s) {
            fr[c]++;
            if (!(fr[c] & 1)) ret += 2;
        }
        return s.size() > ret ? ret + 1 : ret;
    }
};